src/main/scala/shine/DPIA/showDPIA.scala

```diff
@@ -0,0 +1,86 @@
+    case PhraseKind => showPhraseType(arg.asInstanceOf[PhraseType])
+    case DataKind => showDataType(arg.asInstanceOf[DataType])
+    case NatKind => showNat(arg.asInstanceOf[Nat])
+    case AddressSpaceKind => showAddressSpace(arg.asInstanceOf[AddressSpace])
+    case AccessKind => showAccess(arg.asInstanceOf[AccessType])
+    case NatToNatKind => showNatToNat(arg.asInstanceOf[NatToNat])
+    case NatToDataKind => showNatToData(arg.asInstanceOf[NatToData])
+  }
+  def showPrim : Primitive[_] => String = {
+    case Comment(s) => s"/* ${s} */"
+    case Seq(c1, c2) => s"{\n${tabulate(showPhrase(c1))};\n${tabulate(showPhrase(c2))}\n}"
+    case New(dt, f) => s"new (exp[${showDataType(dt)}] x acc[${showDataType(dt)}]) in ${showPhrase(f)}"
+    case Assign(dt, lhs, rhs) => s"${showTypedPhrase(DataKind, dt, lhs)} = ${showPhrase(rhs)}"
+    case IdxAcc(n, dt, idx, array) => s"${showTypedPhrase(DataKind, ArrayType(n, dt), array)}[${showPhrase(idx)}]"
+    case Idx(n, dt, idx, array) => s"${showTypedPhrase(DataKind, ArrayType(n, dt), array)}[${showPhrase(idx)}]"
+    case NatAsIndex(n, e) => s"${showTypedPhrase(NatKind, n, e)}"
+    case Drop(n, m, dt, e) => s"drop ${showNat(n)} ${showTypedPhrase(DataKind, ArrayType(n + m, dt), e)}"
+    case DropAcc(n, m, dt, e) => s"drop ${showNat(n)} ${showTypedPhrase(DataKind, ArrayType(n + m, dt), e)}"
+    case Take(n, m, dt, e) => s"take ${showNat(n)} ${showTypedPhrase(DataKind, ArrayType(n + m, dt), e)}"
+    case TakeAcc(n, m, dt, e) => s"take ${showNat(n)} ${showTypedPhrase(DataKind, ArrayType(n + m, dt), e)}"
+    case f : For => s"for ${showNat(f.n)}\n${tabulate(showPhrase(f.loopBody))}\n"
+    case f : ForNat => s"forNat ${showNat(f.n)}\n${tabulate(showPhrase(f.loopBody))}\n"
+    case p => p.toString
+  }
+
+  def showKindedType[T](kind : Kind[T, _], t : T) : String = showTypes match {
+    case On => s"(${showTypeDo(kind, t)} : ${showKind(kind)})"
+    case Off => showTypeDo(kind, t)
+  }
+  def showTypedPhrase[T,U <: PhraseType](kind : Kind[T, _], t : T, phrase : Phrase[U]) : String = showTypes match {
+    case On => s"(${showPhrase(phrase)} : ${showTypeDo(kind, t)})"
+    case Off => showPhrase(phrase)
+  }
+  def showPhrase[T <: PhraseType] : Phrase[T] => String = {
+    case Identifier(name, t) => if (showTypes == On) s"(${name} : ${t})" else name
+    case Lambda(param, body) => s"λ${showPhrase(param)}. ${showPhrase(body)}"
+    case Apply(fun, arg) => s"${showPhrase(fun)} ${showPhrase(arg)}"
+    case DepLambda(kind, x, body) => s"Λ${showKindedType(kind, x)}. ${showPhrase(body)}"
+    case DepApply(kind, fun, arg) => s"${showPhrase(fun)} ${showKindedType(kind, arg)}"
+    case LetNat(binder, defn, body) => s"let ${showTypeId(binder.id)} = ${showPhrase(defn)} in ${showPhrase(body)}"
+    case PhrasePair(fst, snd) => s"(${showPhrase(fst)}, ${showPhrase(snd)})"
+    case Proj1(pair) => s"fst(${showPhrase(pair)})"
+    case Proj2(pair) => s"snd(${showPhrase(pair)})"
```

Comment:
`(fst ..)` and `(snd ..)` for consistency?

Reply:
done